masonry/src/render_root.rs

```diff
@@ -234,15 +290,38 @@ impl RenderRoot {
         res
     }
 
-    // --- MARK: POINTER_EVENT  ---
+    pub fn get_widget(&self, id: WidgetId) -> Option<WidgetRef<dyn Widget>> {
+        let (state, state_token) = self.widget_arena.widget_states.find(id.to_raw())?;
+        let (widget, widget_token) = self
+            .widget_arena
+            .widgets
+            .find(id.to_raw())
+            .expect("found state but not widget");
+
+        let widget = widget.as_any().downcast_ref::<Box<dyn Widget>>().unwrap();
```

Comment:
This is confusing to me - this seems to from `&Box<dyn Widget>` returned from the arena. Is the arena storing `Box<Box<dyn Widget>> as Box<dyn Widget>`?

Reply:
> Is the arena storing `Box<Box<dyn Widget>> as Box<dyn Widget>`?

It is, and no, that's not ideal. Long term the plan would be to remove a lot of uses of `Box<dyn Widget>` and remove this downcast.

This probably deserves a comment in the code.

Reply:
After some testing, I'm realizing I was confused about what some of the downcasts were doing, and which were necessary. I'm adjusting the code to focus less on the box-of-box aspect, but it's still going to be a bit confusing. I'm adding documentation too.